Dump documentation (#468)
meilisearch/documentation: references/backup.md

# Backup

> The `backup` route allow user to trigger a backup creation and download created backup.

Reviewer: ```suggestion
The `backup` route allow users to trigger a backup creation and downloads the created backup.
```

## Trigger backup

<RouteHighlighter method="POST" route="/backup"/>

> Trigger a backup creation process

Reviewer: ```suggestion
Trigger the backup creation process.
```
When using `a` you imply that there are more than one type of `backup creation processes`. Which I don't think is the case. Correct my if i'm wrong.

### Example

<code-samples id="post_backup_1" />

#### Response: `202 Accepted`

```json
{
    "id": 5
}
```

> ## Get backup status

Reviewer: Why is this not a common `update` with an `updateId` ? Or like @curquiza asked, the ID that is given back on trigger an `updateId` ?
In which case this route seems a bit redundant.

<RouteHighlighter method="GET" route="/backup/:backup_uid/status"/>

> Get backup status.   

Reviewer: ```suggestion
Get the status of a backup.
```

Returned status could be:

> - `processing`: backup creation is not finished
> - `backup_process_failed`: an error occured during backup process, task aborted
> - `done`: backup creation is finished

Reviewer: ```suggestion
- `processing`: Backup creation is not finished.
- `backup_process_failed`: An error occured during backup process, task aborted.
- `done`: Backup creation is finished.
```

Side question: `backup_process_failed` this status can only appear when creating a backup or also when downloading a backup?

### Example

<code-samples id="get_backup_status_1" />

#### Response: `200 Ok`

```json
{
    "status": "processing"
}
```

## Download backup (probably not on first version)

<RouteHighlighter method="GET" route="/backup/:backup_uid"/>

> Download backup.

Reviewer: ```suggestion
Download a compressed backup in `tar.gz` format.
```
